design/gateway/gateway_caching.md

```diff
@@ -0,0 +1,85 @@
+# Ten Gateway Caching Design
+
+## 1. Why cache requests on Gateway?
+
+Currently, all `eth_` requests that hit the gateway are forwarded to our Sequencer. 
+This is not ideal since there is only one Sequencer and it can be overloaded with requests
+and there can be timeouts or errors in the sequencer.
+
+To solve this problem, we can cache the responses of the `eth_` requests on the Gateway.
+Not all requests can be cached, that is why I analyzed the Ethereum JSON-RPC Specification and formed two groups of requests that can be cached and those that cannot be cached.
+For some methods, I used a rule of thumb since results can be cached for a certain period of time, but quickly become outdated.
+
+
+Cacheable request methods are:
+
+- `eth_accounts`
+- `eth_chainID`
+- `eth_coinbase`
+- `eth_getBlockByHash`
+- `eth_getBlockByNumber`
+- `eth_getBlockRecipients`
+- `eth_getBlockTransactionCountByHash`
+- `eth_getBlockTransactionCountByNumber`
+- `eth_getCode`
+- `eth_getStorageAt (Not used)`
+- `eth_getTransactionByBlockHashAndIndex`
+- `eth_getTransactionByBlockNumberAndIndex`
+- `eth_getTransactionByHash`
+- `eth_getTransactionReceipt`
+- `eth_getUncleCountByBlockHash`
+- `eth_getUncleCountByBlockNumber`
+- `eth_maxPriorityFeePerGas`
+- `eth_sign`
+- `eth_signTransaction`
+
+Non-cacheable request methods are:
```

Comment:
they are also cache-able, but with a TTL until the next batch.

Reply:
Fixed. What do you think about `eth_sendRawTransaction` we can probably also catch with TTL until next batch.
(Assuming the key is the whole transaction and submitting the same tx twice in short time period should give the same response).